modo/api.py

```diff
@@ -71,12 +71,13 @@ def __init__(
         self.htsget_endpoint = htsget_endpoint
         self.path = Path(path)
         if s3_endpoint:
-            fs = s3fs.S3FileSystem(endpoint_url=s3_endpoint, **s3_kwargs)
+            s3_opts = s3_kwargs or {"anon": True}
+            fs = s3fs.S3FileSystem(endpoint_url=s3_endpoint, **s3_opts)
             if fs.exists(str(self.path / "data.zarr")):
-                s3_kwargs["endpoint_url"] = s3_endpoint
+                zarr_s3_opts = s3_opts | {"endpoint_url": s3_endpoint}
```

Comment:
__Praise__: Good point and clarification! Also elegant way to concatenate options :+1: